Updated README for Contracts-v1.5.0

Nalon opened the pull request:
This PR updates the Chainlink Smart Contracts README to improve clarity, add structured setup instructions, and align documentation with the latest package changes (v1.5.0). The goal is to make installation, configuration, and usage across different Solidity development environments more straightforward.

Changes to README.md:

- Notes about unvendoring and abi directory changes in v1.5.0
- Includes supporting information around the use of remappings
- Includes setup instructions by environment (Foundry, Hardhat 3, Hardhat 2, Remix)
- Links to starter kits for working examples

Nalon committed:
Updated README for v1.5.0

Updated instructions for v1.5.0.

v1.5.0 unvendors a number of contracts, which require the use of remappings to resolve import paths of package dependencies. Included instructions and information to support this.

- Foundry
- Hardhat 3
- Hardhat 2
- Remix
- Supporting information

zeuslawyer committed:
Minor updates to README. Update ToC

Nalon committed:
Updated HH starter kit information

- Updated HH3 starter kit url to hardhat-3 branch
- Updated link to remapping section of HH SK readme (HH2)
- Added line for HH2 working examples for consistency
- Minor phrasing changes

RensR commented on contracts/README.md:
```diff
@@ -1,25 +1,59 @@
-Add `@chainlink/contracts/=lib/smartcontractkit/chainlink-evm/contracts/` in remappings.txt.
+## Setup & Installation
+
+For use in Solidity projects, further configuration is required. 
```
nit: Can remove this line as it doesn't add much

RensR commented on contracts/README.md:
```diff
@@ -1,25 +1,59 @@
-> When installing via git, the ref defaults to master when no tag is given.
+## NPM Quick Install
 
+> [⚠️ __NOTE__ ]  
```
Why was this syntax chosen? The previous is markdown which renders nicely as a real tag, not just text, although not in NPM. This readme is also the readme in this repo and Git does render the previous tags nicely

RensR commented on contracts/README.md:
````diff
@@ -30,40 +64,156 @@ $ pnpm add @chainlink/contracts
 ```
 
-Add `@chainlink/contracts/=node_modules/@chainlink/contracts/` in remappings.txt.
+#### If you don't want to use `npm`...
````
nit: remove the `...`

RensR commented on contracts/README.md:
```diff
@@ -30,40 +64,156 @@ $ pnpm add @chainlink/contracts
-Add `@chainlink/contracts/=node_modules/@chainlink/contracts/` in remappings.txt.
+#### If you don't want to use `npm`...
+
+If you wish to utilize `forge install`, you will need to install this package as a Git submodule in your project,  using its GitHub URL. This would put this package inside your Foundry project's `./lib` folder.  Please see the [Foundry starter kit](https://github.com/smartcontractkit/foundry-starter-kit).  Note also the use of `remappings.txt` in the [Foundry starter kit](https://github.com/smart ...
```
This makes it sounds like you first have to add it as a submodule before calling forge install. Actually, forge install is installing it as submodule. Maybe we can get away with fewer words here and just tell them to forge install without the other text?

RensR commented on contracts/README.md:
```diff
@@ -30,40 +64,156 @@ $ pnpm add @chainlink/contracts
+
+Here is the [official guidebook](https://getfoundry.sh/guides/project-setup/dependencies) on how to use `forge install` with remappings.
+
+### Step 2: Tell Forge to look in `node_modules`
```
This isn't required if you use remappings right? This is mostly to auto-resolve if you don't use remappings and use the real npm package names in imports

RensR commented on contracts/README.md:
````diff
@@ -30,40 +64,156 @@ $ pnpm add @chainlink/contracts
+
+### Step 4: Usage
+
+```solidity
````
This is the same for all so probably best to not duplicate it for every method

Nalon committed:
Resolved feedback for #244

- Fixed markdown tags
- Updated forge install text to reference forge starter kit
- Removed libs step for foundry
- Removed duplicate usage steps
